apps/ex_shop/web/models/user.ex

```diff
@@ -6,37 +6,49 @@ defmodule ExShop.User do
     field :email, :string
     field :encrypted_password, :string
     field :password, :string, virtual: true
+    field :password_confirmation, :string, virtual: true
     field :is_admin, :boolean
 
     timestamps
   end
 
-  #@required_fields ~w(name email encrypted_password)
-  #@optional_fields ~w()
+  def changeset(model, params \\ :empty) do
+    model
+    |> cast(params, ~w(), ~w())
+  end
 
   @doc """
-  Creates a register form changeset based on the `model` and `params`.
-
-  If no params are provided, an invalid changeset is returned
-  with no validation performed.
+  Refer Registration changeset
   """
-  def register_admin_form_changeset(model, params \\ :empty) do
-    model
-    |> cast(params, ~w(name email password is_admin), ~w())
-  end
+  def create_changeset(model, params \\ :empty), do: changeset(model, params) |> add_error(:email, "Refer Registration changeset")
 
   @doc """
-  Creates a create user changeset based on the `model` and `params`.
+  Updates a user changeset based on the `model` and `params`.
 
   If no params are provided, an invalid changeset is returned
   with no validation performed.
   """
-  def create_changeset(model, params \\ :empty) do
+  def update_changeset(model, params \\ :empty) do
     model
-    |> cast(params, ~w(name email password), ~w())
+    |> cast(params, ~w(email), ~w(name password password_confirmation is_admin))
+    |> changeset_helper
+  end
+
+  def changeset_helper(changeset) do
+    changeset
     |> validate_confirmation(:password, message: "password does not match")
     |> validate_format(:email, ~r/@/)
     |> validate_length(:password, min: 5)
     |> unique_constraint(:email)
+    |> set_hashed_password
+  end
+
+  defp set_hashed_password(changeset = %{errors: [_]}), do: changeset
```

Comment:
can we pattern match on valid? instead. if in future there are nested assocs here, this won't reflect the errors, also will skip if more than one errors.